src/mc_cell_group.hpp

```diff
@@ -124,6 +102,34 @@ class mc_cell_group: public cell_group {
 
         lowered_.setup_integration(tfinal, dt);
 
+        // Set up sample event queue.
+        struct sampler_entry {
+            typename lowered_cell_type::probe_handle handle;
+            probe_tag tag;
+            cell_member_type probe_id;
+            sampler_function sampler;
+        };
+        std::vector<sampler_entry> samplers;
```

Comment:
It would be better heap-behaved, with a small cost in implementation complexity. But all this code will be reworked when the lowered cell sampling implementation is done, so perhaps best addressed there?

Reply:
Fine.